### ISSUE TITLE
can not install evcxr_jupyter  under macos 

### ISSUE THREAD

**Issue by minikiller** [google/evcxr#229]
```
 Compiling ra_ap_project_model v0.0.104
error[E0599]: no method named `parse` found for reference `&cargo_metadata::Edition` in the current scope
   --> /Users/sunlingfeng/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/ra_ap_project_model-0.0.104/src/cargo_workspace.rs:319:35
    |
319 |             let edition = edition.parse::<Edition>().unwrap_or_else(|err| {
    |                                   ^^^^^ method not found in `&cargo_metadata::Edition`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `ra_ap_project_model` due to previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `evcxr_jupyter v0.12.0 (https://github.com/google/evcxr.git#cc140009)`, intermediate artifacts can be found at `/var/folders/my/x1pz96mj4_l31xr74gb254fh0000gn/T/cargo-installXUDWhB`
```

**Comment by davidlattimore** [comment]
So this error was with an earlier version of rust? Do you get the same error now that you've upgraded to 1.61.0?

**Comment by davidlattimore** [comment]
I just did some more investigation, trying to figure out what's going on. It looks like the variable `edition` that `.parse` is being called on is a String for me, but it's an `&cargo_metadata::Edition` for you. It seems that the type of that field was changed from version 0.14 to version 0.15 of cargo_metadata. `Cargo.toml` for version `0.0.104` of the crate `ra_ap_project_model` requests version `0.14.2`. AFAIK, Cargo's semver rules mean that it shouldn't select version 0.15 when version 0.14 was requested. Not sure why you're ending up with 0.15. Perhaps if 0.14 isn't available for some reason, it'll pick a higher version. It looks like you're using a mirror for crates.io, so perhaps that's related - possibly the mirror doesn't have cargo_metadata version 0.14.2.

**Comment by bjorn3** [comment]
> Perhaps if 0.14 isn't available for some reason, it'll pick a higher version.

No, in that case cargo should either give an error or try to pick a different version of `ra_ap_project_model` for which it can find all dependencies.

**Comment by davidlattimore** [comment]
Thanks @bjorn3. Actually, another thing that seems surprising is that it's using ra_ap_project_model v0.0.104 - that's what's used by the evcxr in github, but I haven't published to crates.io in a while and the version of ra_ap_project_model used by evcxr on crates.io is "=0.0.76".

@minikiller - do you have any ideas why different versions of packages might be getting pulled in when you run `cargo install`? Perhaps you can provide more detail about your setup and how you're installing stuff?

**Comment by bjorn3** [comment]
> that's what's used by the evcxr in github, but I haven't published to crates.io in a while

The last line shows that @minikiller was compiling directly from the git repo as of commit cc140009.

**Comment by davidlattimore** [comment]
I just released 0.13.0 to crates.io. It includes a significant bump to the rust-analyzer version. The new version uses cargo_metadata version 0.15, so this build failure, however it was happening should hopefully be fixed now.

**Comment by bitsnaps** [comment]
I got the same issue `error: failed to compile `evcxr_jupyter v0.13.0`, intermediate artifacts can be found at `/var/folders/...`, I believe this has to do with xcode which is not installed/updated in my machine at the moment.
